bgpd/bgp_vty.c

```diff
@@ -6692,6 +6692,28 @@ bgp_show_summary (struct vty *vty, struct bgp *bgp, int afi, int safi,
   return CMD_SUCCESS;
 }
 
+/*
+ * Return the # of peers that are configured to use this afi/safi
+ */
+static int
+bgp_show_summary_afi_safi_peer_count (struct bgp *bgp, int afi, int safi)
```

Comment:
since you really only want to know if there are no neighbors, there's no need to walk the whole list.  The function could be changed to something like bgp_show_summary_afi_safi_have_neighbor and return 0 for none and 1 for !=0, and not walk what could eventually be a long list.